components/sup/src/manager/service/mod.rs

```diff
@@ -387,29 +392,33 @@ impl Service {
         self.supervisor.state == ProcessState::Down
     }
 
-    /// Compares the current state of the service to the current state of the census ring and
-    /// re-renders all templatable content to disk.
+    /// Compares the current state of the service to the current state of the census ring and the
+    /// user-config, and re-renders all templatable content to disk.
     ///
-    /// Returns true if any modifications were made.
+    /// Returns `true` if any modifications were made.
     fn update_templates(&mut self, census_ring: &CensusRing) -> bool {
         let census_group = census_ring.census_group_for(&self.service_group).expect(
             "Service update failed; unable to find own service group",
         );
-        let cfg_updated = self.cfg.update(census_group);
+        let cfg_updated = self.cfg.update(census_group) || self.user_config_updated;
         if cfg_updated || census_ring.changed {
+            if let Err(e) = self.cfg.load_user(&self.pkg) {
+                outputln!(preamble self.service_group, "Loading user-config failed: {}", e);
+            }
             let (reload, reconfigure) = {
                 let ctx = self.render_context(census_ring);
-                let reload = self.compile_hooks(&ctx);
+                let reload = self.compile_hooks(&ctx) || self.user_config_updated;
```

Comment:
Should this `|| self.user_config_updated` be moved one line below, to the `let reconfigure` line?

Reply:
No, the `needs_reconfiguration` flag is only used to [trigger a `reconfigure` hook](https://github.com/habitat-sh/habitat/blob/c827ae61fa9077974e22fdbf04aecbafefbcef19/components/sup/src/manager/service/mod.rs#L639-L646), which may not be defined. The `reload` flag is actually going to trigger a service reload, which is what we need here.